pageserver/src/tenant/storage_layer/delta_layer.rs

```diff
@@ -610,18 +612,28 @@ impl Drop for DeltaLayerWriter {
 }
 
 impl DeltaLayerInner {
+    /// Returns nested result following Result<Result<_, OpErr>, Critical>:
+    /// - inner has the success or transient failure
+    /// - outer has the permanent failure
     pub(super) async fn load(
         path: &Utf8Path,
         summary: Option<Summary>,
         ctx: &RequestContext,
-    ) -> anyhow::Result<Self> {
-        let file = VirtualFile::open(path)
-            .await
-            .with_context(|| format!("Failed to open file '{path}'"))?;
+    ) -> Result<Result<Self, anyhow::Error>, anyhow::Error> {
+        let file = match VirtualFile::open(path).await {
+            Ok(file) => file,
+            Err(e) => return Ok(Err(anyhow::Error::new(e).context("open layer file"))),
+        };
```

Comment:
```suggestion
        let file = VirtualFile::open(path).await.map_err(|e| anyhow::Error::new(e).context("open layer file"))?;
```

Reply:
This suggestion would return: `Err(...)`, but I want to return `Ok(Err(...))`.

The question-mark does not really support this kind of error nesting. I wished it did. I failed to see a nice solution with a custom error type either, because then you'd need `either::Either` as the summaries are different types between delta and image.